File: test/python/MultiAttribute.py
#!/usr/bin/env python2
import sys
import hyperdex.client
from hyperdex.client import LessEqual, GreaterEqual, LessThan, GreaterThan, Range, Regex, LengthEquals, LengthLessEqual, LengthGreaterEqual
c = hyperdex.client.Client(sys.argv[1], int(sys.argv[2]))
def to_objectset(xs):
    return set([frozenset(x.items()) for x in xs])
assert c.get('kv', 'k') == None
assert c.put('kv', 'k', {'v1': 'ABC'}) == True
assert c.get('kv', 'k') == {'v1': 'ABC', 'v2': ''}
assert c.put('kv', 'k', {'v2': '123'}) == True
assert c.get('kv', 'k') == {'v1': 'ABC', 'v2': '123'}
assert c.get_partial('kv', 'k', ['v1']) == {'v1': 'ABC'}

File: test/ruby/MultiAttribute.rb
#!/usr/bin/env ruby
require 'hyperdex'

def assert &block
    raise RuntimeError unless yield
end

def collapse_iterator(xs)
    s = Set.new []
    while xs.has_next() do
        x = xs.next()
        x.freeze
        s.add(x)
    end
    s
end

def to_objectset(xs)
    s = Set.new []
    xs.each do |x|
        x.freeze
        s.add(x)
    end
    s
end

c = HyperDex::Client::Client.new(ARGV[0], ARGV[1].to_i)
assert { c.get("kv", "k") == nil }
assert { c.put("kv", "k", {"v1" => "ABC"}) == true }
assert { c.get("kv", "k") == {:v1 => "ABC", :v2 => ""} }
assert { c.put("kv", "k", {"v2" => "123"}) == true }
assert { c.get("kv", "k") == {:v1 => "ABC", :v2 => "123"} }
assert { c.get_partial("kv", "k", ["v1"]) == {:v1 => "ABC"} }
